### DISCUSSION TITLE
appium:mjpegScreenshotUrl - is there a video equivalent?

### DISCUSSION THREAD

**Discussion by tking16** [appium/appium#20002] [Q&A]
Two quick questions regarding using mjpeg capabilities when using Selenium Grid, 
1) Following this [guide](https://appium.io/docs/en/2.1/guides/grid/) I added `appium:mjpegScreenshotUrl: "http://localhost:9100"` to my 6 yaml files, but I record failing videos too, is there a `mjpegVideoUrl` capability or something for this? Couldn't find anything online
2) 3 of my Appium Simulators are being run on remote macmini Node, do I keep `mjpegScreenshotUrl` as `localhost` or does this become the ip address of the Selenium Grid hub? The screenshots are needed for the html-report.

Example of one my remote appium.yml files
```
server:
  port: 4743
  use-drivers:
    - xcuitest
  default-capabilities:
    appium:wdaLocalPort: 8102
    appium:derivedDataPath: ~/nodeData6
    appium:wdaLaunchTimeout: 720000 # 2 minutes per Simulator
    #appium:usePrebuiltWDA: true
    appium:settings[pageSourceExcludedAttributes]: "visible"
    appium:newCommandTimeout: 9999
    appium:isHeadless: true
    appium:mjpegServerPort: 9116
    appium:mjpegScreenshotUrl: "http://localhost:9116"
```

**Comment by portyu9** [comment]
Hi @tking16 

There is no `mjpegVideoUrl` capability.

For video, use Appium's screen-recording API:

```java
driver.startRecordingScreen();

// run test

String video = driver.stopRecordingScreen();
```

The XCUITest driver can use its MJPEG stream internally to create the recording.

For:

```yaml
appium:mjpegScreenshotUrl: "http://localhost:9116"
```

keep `localhost` if Appium and the forwarded MJPEG port are on the **same remote Mac node**. The URL must be reachable from the Appium process, not from the Selenium Grid hub.

Only use the node's IP/hostname instead if some external machine needs to connect directly to that MJPEG stream.

Please mark this answer as accepted if it helped, thank you.
